### DISCUSSION TITLE
Not able to fetch the content list from Knowlg Platform

### DISCUSSION THREAD

**Discussion by Snegha-M** [Sunbird-Knowlg/Community#114] [General Discussions]
Hi Team 
Content Player: https://github.com/Sunbird-Knowlg/sunbird-content-player/tree/release-4.1.1
Knowlg Platform: https://github.com/Sunbird-Knowlg/knowledge-platform/tree/release-4.8.0
Successfully we can upload a pdf content file through API using Sunbird Knowlg, now we want to see the content of the pdf file which we had uploaded, So we have selected Content player UI to see the content of the pdf file and we were able to set up that content player locally in our machine. 
1. We altered the configuration by adding the URL of the backend service in the appConfig.js file of the content player.
![Screenshot from 2023-03-24 16-41-43](https://user-images.githubusercontent.com/99195190/227506564-6e715f6d-009f-437e-afed-b8723b251b6c.png)

2. We came to know that the taxonomy service has been used in the content player but whatever the endpoint used there doesn't match with the endpoint which is listed in Sunbird Knowlg.
![Screenshot from 2023-03-24 16-38-50](https://user-images.githubusercontent.com/99195190/227506302-af6c4c1a-f3b3-4636-bd88-dcadce76dfe9.png)

Please guide us how to get the content list.

**Comment by HarishGangula** [comment]
@Snegha-M  For playing the pdf content you can use https://github.com/Sunbird-Knowlg/sunbird-pdf-player/tree/release-5.3.0 and there are no taxonomy service details and changes required for this.

**Comment by Snegha-M** [comment]
@HarishGangula Can you please help us to understand how to fetch the list of content which was uploaded using Knowlg content service upload API?

**Reply by pallakartheekreddy** [reply]
@Snegha-M Please use the `search service` from Knowlg to get the list of contents what you upload. 
https://knowlg.sunbird.org/learn/product-and-developer-guide/assets-search-service

**Comment by Snegha-M** [comment]
@pallakartheekreddy In content player, we are able to see that data are hard-coded, is there any existing code or repo which sunbird use to display/view/consume dynamic content ( like how Diksha is working ) or do I need to write custom code using the search service ?

Please let us know if you have any solution for displaying the dynamic content.


The search API returns response like below
{
    "id": "api.search-service.search",
    "ver": "3.0",
    "ts": "2023-03-28T17:55:55Z+05:30",
    "params": {
        "resmsgid": "bfab1cb7-3547-4a86-ab0f-60c7f0f43bcd",
        "msgid": null,
        "err": null,
        "status": "successful",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "count": 0
    }
}

There are no details about Logstash and Flink in Sunbird Knowlg Documentation.
The following screenshot represents the architecture that we found in the sunbird Community discussions. According to the diagram we observe that Logstash and Flink job should be set up.

![Screenshot from 2023-03-28 15-46-42 (1)](https://user-images.githubusercontent.com/99195190/228236298-41fc3b5a-743e-4cde-a0f7-fc096fa0b056.png)

Could you please help us to set up Logstash and Flink job?

**Comment by Snegha-M** [comment]
@pallakartheekreddy **Can you please provide a solution**

**Comment by maheshkumargangula** [comment]
@Snegha-M – The local setup documentation for Logstash and Flink jobs is in progress. Util that time, please use the below Ansible roles to understand deployment steps.

- **logstash**: Please use the configuration from [logstash-deploy](https://github.com/Sunbird-Knowlg/sunbird-learning-platform/tree/release-5.4.0/ansible/roles/logstash-deploy) Ansible role and deploy it. 
- **Flink Jobs**: You can follow the standard Flink jobs deployment process to deploy the Flink jobs. Please check the config file inside resources folder of the specific job and make the required changes before building the code.

We will share the local setup documentation details here once they are complete and ready.
Cc: @pallakartheekreddy
